controllers/admin/AdminQualityAssuranceController.php

```diff
@@ -197,6 +204,14 @@ public function ajaxProcessGetHookCallLogs()
         $this->renderJson($grouped);
     }
 
+    public function ajaxProcessToggleJSEventListener()
+    {
+        $oldStatus = Configuration::get('PS_QA_MODULE_LISTEN_JS');
+        $result = Configuration::updateValue('PS_QA_MODULE_LISTEN_JS', ($oldStatus ? 0 : 1));
```

Comment:
```suggestion
        $result = Configuration::updateValue('PS_QA_MODULE_LISTEN_JS', (bool) $oldStatus);
```

Reply:
I see no reason to save it in integer, then treat it as boolean

Reply:
Because in database it'll become an integer. I prefer to be clear with the reader about the nature of this data once stored in database.

I prefer to make unambigous to reader the data storage is flawed.

Reply:
Regarding your comment, we will never be able to use boolean because MySQL transforms boolean in TinyInt :scream: 
If you want to match the database schema, use string, not integer :)

Reply:
> Regarding your comment, we will never be able to use boolean because MySQL transforms boolean in TinyInt 😱

Yes, we will be able to do it if we use Doctrine. Because when you give a boolean to Doctrine, it formats it into TinyInt for MySQL, but **what is the key** is that when it fetches the TinyInt again it gives you again a boolean.

**Doctrine**

- you give it a boolean to store
- when you fetch it ; it gives you back a boolean

**ObjectModel**

- you give it a boolean to store
- when you fetch it ; it gives you back an integer

So rather than following the inconsistent "boolean into integer" path of ObjectModel I prefer to stay with integers. At least I'm consistent all the way.

Reply:
It's the Configuration ObjectModel, the value field is in text format so it's a string, not an integer.
I'm afraid you have to refactor everything you did with all Configuration calls in the Core with your reflection :sweat_smile: 

Anyway, the ObjectModel is broken, doesn't mean we have to follow it, we should use what it makes sense, then remove the cast when we will switch to doctrine.

Reply:
I approve @PierreRambaud. We must follow good practices. And not practices hidden by ObjectModel. If the value must be a boolean, we must cast it to boolean.

Reply:
> It's the Configuration ObjectModel, the value field is in text format so it's a string, not an integer.

Hoho 😄  you're right it's even worse than I thought.

Reply:
I know, I'm always right :trollface:

Reply:
You're on the way to become a 10x developer: you have the right ego :trollface:

Reply:
I have the ego for 20x developer.

Reply:
<img src="https://pbs.twimg.com/media/DijHqIdW0AA6mKY.jpg">

Reply:
![](https://pics.awwmemes.com/php-sucks-inngfip-com-19115238.png)

Reply:
<img src="https://i.imgur.com/ujBFBJl.jpg">

Reply:
![](https://i.imgflip.com/4psouw.jpg)

Reply:
![](https://i.chzbgr.com/full/1483445504/hA3B81DBB/im-ready-let-the-war-begin)

Reply:
<img src="https://res.cloudinary.com/practicaldev/image/fetch/s--edliSUQ3--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://cdn-images-1.medium.com/max/400/0%2A1WlehGOk0peCQhbx.jpg">

Reply:
![](https://memegenerator.net/img/instances/32765121/nice-try-apple-fag.jpg)